### DISCUSSION TITLE
JWT creating erros

### DISCUSSION THREAD

**Discussion by MatD1** [nextauthjs/next-auth#2990] [Help]
I am trying to setup JWT with NextAuth and it's just being a pain, reading through posts on GitHub, and researching is returning no help. I have a feeling it's some error happening when NextAuth attempts to decode the JWT payload, I am just trying to return a string of numbers that are returned from Twitch upon signin, I need this to use in a API call to Twitch. 

I am new to JWT and NextAuth so I am no pro! 

github repo for entire project https://github.com/MatWebsites/next-auth-example-main
### Any help would be huge!

```console
// console errors
unhandledRejection: TypeError: Cannot read properties of undefined (reading 'next-auth.session-token')
-
error - Error: Objects are not valid as a React child (found: [object Promise]). If you meant to render a collection of children, use an array instead.
-
TypeError: getCurves is not a function
-

```
nextjs error page [![error][1]][1]

```javascript
//[...nextauth].js
import NextAuth from "next-auth"
import Providers from "next-auth/providers"

export default NextAuth({
  providers: [
    Providers.Twitch({
      clientId: process.env.TWITCH_CLIENT_ID,
      clientSecret: process.env.TWITCH_CLIENT_SECRET,
    }),
    Providers.Spotify({
      clientId: process.env.SPOTIFY_CLIENT_ID,
      clientSecret: process.env.SPOTIFY_CLIENT_SECRET
    }),
  ],
  secret: process.env.SECRET,

  session: {
    jwt: true,
  },

  // https://next-auth.js.org/configuration/options#jwt
  jwt: {
    // A secret to use for key generation (you should set this explicitly)
     secret: process.env.SECRET,
    // Set to true to use encryption (default: false)
    // encryption: true,
    // You can define your own encode/decode functions for signing and encryption
    // if you want to override the default behaviour.
    // encode: async ({ secret, token, maxAge }) => {},
    // decode: async ({ secret, token, maxAge }) => {},
  },
  callbacks: {
     async signIn(user, account, profile) { return true },
    // async redirect(url, baseUrl) { return baseUrl },
     async session(session, user) { return session },
     async jwt(token, user, account, profile, isNewUser) { 
       token.accessToken = account.access_token;
       return token 
      }
  },
  theme: 'light',

  debug: true,
})
```
```javascript
//test.js - trying to decode and use the data inside the web token
import { getToken } from 'next-auth/jwt'
const secret = process.env.SECRET

const test =  (req, res) => {
    const token = getToken({req, secret});
    return (
        <div>
            <a>{token.sub}</a> // Should show the users id returned from Twitch
        </div>
    )
}

export default test
```

  [1]: https://i.stack.imgur.com/l73ZN.png

**Comment by balazsorban44** [comment]
The repo is private, so I cannot have a look. The error logs indicate no connection to next-auth. The example code you link to makes it unclear if you are in an API route or trying to render a page, but in any case you are doing something wrong.

getToken is a server-side, async method so you cannot use it in page components. Use the `useSession` hook instead. In case if you are in an API route, you cannot return JSX either.

The problem is in your side in any case.

**Reply by MatD1** [reply]
Ah yeah, I've changed the repo to public, but my mistake was trying to use this in a page I realised this cannot be done, I need to somehow get the users ID that is received from Twitch upon sign-In, everything I have tried has not worked, I thought using JWT may have been a better idea, but I do not think it is.

**Reply by balazsorban44** [reply]
It is. check out the jwt and session callbacks https://next-auth.js.org/configuration/callbacks

**Reply by MatD1** [reply]
Yeah I am attempting to figure this out, but after adding the signing key and secret im just getting this error upon trying to sign in 
`[next-auth][error][OAUTH_CALLBACK_HANDLER_ERROR] 
https://next-auth.js.org/errors#oauth_callback_handler_error Unexpected token k in JSON at position 0 {
  message: 'Unexpected token k in JSON at position 0',
  stack: 'SyntaxError: Unexpected token k in JSON at position 0\n' +
`

**Reply by rishabhtatia1** [reply]
I am to getting the same error.
Did you find the solution?

**Reply by sofiaFern** [reply]
also getting the same error
